### ISSUE TITLE
Add support for Microsoft SQL Server

### ISSUE THREAD

**Issue by ryano144** [erezsh/reladiff#20]
Although it wasn't great, there was basic support added for MS SQL in the main data-diff back in September in this PR: https://github.com/datafold/data-diff/pull/696/.

Then updated slightly in this one: https://github.com/datafold/data-diff/pull/825

**Comment by erezsh** [comment]
I am aware that a MS-SQL driver has been attempted.

@ryano144  Have you used and benchmarked their implementation?

My impression is that the driver is too slow to be useful for cross-db comparison, as it uses the `HashBytes()` function that it is approx 100 times slower than equivalent function in postgres .

My perhaps there are details that I'm not aware of?

**Comment by boldandbusted** [comment]
Is it possible to still add support even if the performance is 100x slower than in postgres? Diffing could be considered a kind of 'batch job'. The results might be worth waiting for overnight! :)

Or, is the performance impact such that the tool just can't handle such a long runtime? I'm excited to find this tool for diffing databases! Cheers. :)

**Comment by erezsh** [comment]
@boldandbusted My impression is that at that point, it would be faster to just download the data. But maybe I'm seeing it wrong?

Also yeah, timeouts are a concern.

**Comment by boldandbusted** [comment]
> @boldandbusted My impression is that at that point, it would be faster to just download the data. But maybe I'm seeing it wrong?
> 
> Also yeah, timeouts are a concern.

I dont think you are seeing it wrong. A guiding criteria for this project is speed and high performace, after all! :) But, there may be a use just in having this utility as a common "controller" even for a slow process. Especially if it is used in an environment with a mixture of database engines.

But, I have no PR to offer, so I know my suggestion is only worth the value of the code I offer. ;).
